### ISSUE TITLE
acl.conf configuration not being read.

### ISSUE THREAD

**Issue by pritukam** [emqtt/emqttd#998]
#### Environment

- OS: Windows
- Erlang/OTP: 19
- EMQ: 2.1.0

#### Description
We have been trying to limit the access control using the acl.conf file.
We tried restricting access to a specific machine using IP address as follows:

{deny, {ipaddr, "172.16.36.255"}, subscribe, ["$SYS/#", "#"]}.

Yet the other machine with the above ip is able to effortlessly subscribe to restricted topics and is receiving messages sent on it.

Our use case wants that only one specified machine be able to subscribe to the topic and no one else.

We tried denying all users, access to subscribe and then allowing this IP to subscribe, that didn't work either.

Our acl.conf file looks like this:
---------------------------------------------------------------------------------------------
{allow, {user, "dashboard"}, subscribe, ["$SYS/#"]}.

{allow, {ipaddr, "127.0.0.1"}, pubsub, ["$SYS/#", "#"]}.

%%{deny, all, subscribe, ["$SYS/#", {eq, "#"}]}.

%%{allow, all}.

{deny, {ipaddr, "172.16.36.255"}, subscribe, ["$SYS/#", "#", "benchmark", {eq, "#"}]}. %%benchmark is the topic in question

------------------------------------------------------------------------------------------------

Please let me know if anything is done wrong. Thanks.

**Comment by emqplus** [comment]
@pritukam The documentation on ACL: http://emqtt.io/docs/v2/guide.html#acl. You have to define '{allow, all}' or '{deny, all}' at the end of acl.conf.

**Comment by pritukam** [comment]
Hi @emqplus, I tried that too. 

{allow, {ipaddr, "172.16.36.149"}, subscribe, "benchmark" }.
{deny, all}.

I added this to the config, hoping that the if I subscribe from the machine with the IP, the messages can be subscribed. I am unable to receive the sent messages from any machine, including the one that is allowed to subscribe and receive.

I tried {allow, all} and any machine is allowed to connect now.  Our use case dictates that only one specified machine be able to subscribe to the topic and no one else.

Is there something I am missing?

p.s. thanks a lot for replying.
